Common/Statistics/AssetsUnderManagementCapacityManager.cs

```diff
@@ -0,0 +1,245 @@
+    public class AssetsUnderManagementCapacityManager
+    {
+        private readonly object _lock = new object();
+
+        private readonly SecurityPortfolioManager _portfolio;
+        private readonly ISubscriptionDataConfigProvider _subscriptionDataConfigProvider;
+
+        private readonly Dictionary<Symbol, decimal> _maximumSymbolCapacity;
+        private readonly RollingWindow<decimal> _historicalPortfolioCapacity;
+        private DateTime _previousInputTime;
+
+        /// <summary>
+        /// Assets Under Management (AUM) Capacity
+        /// </summary>
+        public decimal AumCapacity =>
+            _historicalPortfolioCapacity.Count == 0 ? 0 : _historicalPortfolioCapacity.Average();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AssetsUnderManagementCapacityManager"/> class
+        /// </summary>
+        /// <param name="portfolio">The algorithm's portfolio</param>
+        /// <param name="subscriptionDataConfigProvider">Provides access to registered <see cref="SubscriptionDataConfig"/></param>
+        /// <param name="orderEventProvider">Provides access to the order events</param>
+        public AssetsUnderManagementCapacityManager(
+            SecurityPortfolioManager portfolio,
+            ISubscriptionDataConfigProvider subscriptionDataConfigProvider,
+            IOrderEventProvider orderEventProvider)
+        {
+            _portfolio = portfolio;
+            _subscriptionDataConfigProvider = subscriptionDataConfigProvider;
+
+            _maximumSymbolCapacity = new Dictionary<Symbol, decimal>();
+            _historicalPortfolioCapacity = new RollingWindow<decimal>(30);
+            _previousInputTime = DateTime.MinValue;
+
+            orderEventProvider.NewOrderEvent += HandleNewOrderEvent;
+        }
+
+        /// <summary>
+        /// Handles new order event:
+        /// When dealing with a security with high resolution data, create a consolidator to compute the
+        /// total trade volume capacity using a volume of the past 5 minutes. For low resolution, use the last bar.
+        /// </summary>
+        private void HandleNewOrderEvent(object sender, OrderEvent orderEvent)
+        {
+            var symbol = orderEvent.Symbol;
+
+            var configs = _subscriptionDataConfigProvider.GetSubscriptionDataConfigs(symbol);
+            if (configs.IsNullOrEmpty())
+            {
+                Log.Error($"AssetsUnderManagementCapacityManager: Could not find any SubscriptionDataConfig for {symbol}.");
+                return;
+            }
+
+            var consolidator = default(IDataConsolidator);
+            var hasHighResolution = false;
+            var security = _portfolio.Securities[symbol];
+            BaseData lastData = security.Cache.GetData<TradeBar>();
+
+            // For high resolution data, create a consolidator that will compute
+            // the AUM Capacity after a 5-minute period bar is closed
```

Comment:
maybe a note here saying that we're only adding the consolidator for the first config -- also, should there be preference of trades over quotes or vice versa? Right now we're just depending on whatever order the provider returns them in which (may) cause non-determinism later (perhaps)?